### PR TITLE
[WIP] Making test/eventrouter.sh more robust.

### PR DESCRIPTION
**Pull request by nhosoi**

### CODE REVIEW

File: test/eventrouter.sh [hunk 1]
```diff
@@ -6,7 +6,7 @@ source "$(dirname "${BASH_SOURCE[0]}" )/../hack/lib/init.sh"
 source "${OS_O_A_L_DIR}/hack/testing/util.sh"
 os::test::junit::declare_suite_start "test/eventrouter"
 
-FLUENTD_WAIT_TIME=${FLUENTD_WAIT_TIME:-$(( 3 * minute ))}
+FLUENTD_WAIT_TIME=${FLUENTD_WAIT_TIME:-$(( 5 * minute ))}
```
Comment (richm) [inline test/eventrouter.sh]: do you have evidence that part of the problem is that it takes fluentd too long to process eventrouter records?

File: test/eventrouter.sh [hunk 2]
```diff
@@ -60,37 +62,57 @@ else
...
+    oc label node --all logging-infra-fluentd- 2>&1 | artifact_out
+    os::cmd::try_until_text "oc get daemonset logging-fluentd -o jsonpath='{ .status.numberReady }'" "0" $FLUENTD_WAIT_TIME
     oc set env ds/logging-fluentd MUX_CLIENT_MODE- 2>&1 | artifact_out
-    start_fluentd false 2>&1 | artifact_out
+    oc label node --all logging-infra-fluentd=true 2>&1 | artifact_out
+    os::cmd::try_until_text "oc get pods -l component=fluentd" "^logging-fluentd-.* Running "
 
     warn_nonformatted $essvc '/project.*'
-    warn_nonformatted $esopssvc '/.operations.*'
+    warn_nonformatted $esopssvc '/.operations.*/'
```
Comment (richm) [inline test/eventrouter.sh]: why would this need the trailing `/` but not `/project.*`?

File: test/eventrouter.sh [hunk 3]
```diff
@@ -60,37 +62,57 @@ else
...
 
-    os::cmd::try_until_not_text "curl_es $esopssvc /.operations.*/_count?q=kubernetes.event.verb:* | get_count_from_json" "^0\$" $FLUENTD_WAIT_TIME
-    prev_event_count=$( curl_es $esopssvc /.operations.*/_count?q=kubernetes.event.verb:* | get_count_from_json )
+    qs='{"query":{"wildcard":{"kubernetes.event.verb":"*"}}}'
+    escapedqs='{\"query\":{\"wildcard\":{\"kubernetes.event.verb\":\"*\"}}}'
+    os::cmd::try_until_success "logs_count_is_gt 0 $escapedqs" $FLUENTD_WAIT_TIME
```
Comment (richm) [inline test/eventrouter.sh]: you should be able to do this:
```
     os::cmd::try_until_success "logs_count_is_gt 0 '$qs'" $FLUENTD_WAIT_TIME
```

**Comment by richm** [comment]
This is about the 3.10 backport?  Not sure why that has so many problems on 3.10 - maybe we missed backporting some test fix?

**Comment by nhosoi** [comment]
> This is about the 3.10 backport? Not sure why that has so many problems on 3.10 - maybe we missed backporting some test fix?

Well, I'm sure not all of the changes were needed just to make the test pass on 3.10.  Some are just for checking and debugging.  I thought it's better to have the same/similar code across the versions.

Thanks for the comments.  I'm going to fix them.  But so far no luck to have the deployment successfully done using the master branch... (both CI and get-machine script)

File: test/eventrouter.sh (hunk 1 above)
Comment (nhosoi) [reply test/eventrouter.sh]: I could reduce the time and rerun the test.  Do you want me to make it shorter in https://github.com/openshift/origin-aggregated-logging/pull/1437, as well?

File: test/eventrouter.sh (hunk 2 above)
Comment (nhosoi) [reply test/eventrouter.sh]: I guess it's not needed.  A leftover in my struggle...  Will remove it.

File: test/eventrouter.sh (hunk 3 above)
Comment (nhosoi) [reply test/eventrouter.sh]: Thanks.  Leg me try...

**Comment by nhosoi** [comment]
Thanks, @richm!  The same eventrouter.sh worked just fine on v3.10.
https://github.com/openshift/origin-aggregated-logging/pull/1437

**Comment by nhosoi** [comment]
rebased with master.

**Comment by nhosoi** [comment]
[CI test failure summary]
**ci/prow/rhel-images**
`error: could not resolve inputs: could not determine inputs for step [input:ruby-25]: could not resolve base image: imagestreamtags.image.openshift.io "4.0:ruby-25" not found`

**ci/openshift-jenkins/json-file**
**ci/openshift-jenkins/logging**
```
TASK [openshift_cli : Pre-pull cli image] **************************************
task path: /usr/share/ansible/openshift-ansible/roles/openshift_cli/tasks/main.yml:13
        "manifest for docker.io/openshift/origin-node:9d2874f not found"
```

**Comment by richm** [comment]
> [CI test failure summary]
> **ci/prow/rhel-images**
> `error: could not resolve inputs: could not determine inputs for step [input:ruby-25]: could not resolve base image: imagestreamtags.image.openshift.io "4.0:ruby-25" not found`

Looks like we need to somehow add that to the rhel7 ci
 
> **ci/openshift-jenkins/json-file**
> **ci/openshift-jenkins/logging**
> 
> ```
> TASK [openshift_cli : Pre-pull cli image] **************************************
> task path: /usr/share/ansible/openshift-ansible/roles/openshift_cli/tasks/main.yml:13
>         "manifest for docker.io/openshift/origin-node:9d2874f not found"
> ```

These are pretty much permanently broken - they will have to be replaced with the new api.ci gcp operator based ci tests when we get them working

**Comment by richm** [comment]
also note that ci/openshift-jenkins/json-file and ci/openshift-jenkins/logging are not blocking jobs for master branch

**Pull request closed without merging**